pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -67,6 +66,13 @@ func (c *Cluster) makeDeployment(mgrConfig *mgrConfig) (*apps.Deployment, error)
 		},
 	}
 
+	// If the log collector is enabled we add the side-car container
+	if c.spec.LogCollector.Enabled {
+		shareProcessNamespace := true
+		podSpec.Spec.ShareProcessNamespace = &shareProcessNamespace
+		podSpec.Spec.Containers = append(podSpec.Spec.Containers, *controller.LogCollectorContainer(fmt.Sprintf("ceph-mgr.%s", mgrConfig.DaemonID), c.clusterInfo.Namespace, c.spec))
+	}
+
```

Comment:
A suggestion: I see we are adding a change to all pods for all daemons. What if we took the time here to create a base Pod template that all daemons could get and then customize for their purposes? We could make it so the template automatically applies appropriate labels to the template as well as a lot of the other things that effectively get copy-pasted. This is also something we could do for code cleanup/refactoring later, but it seems like it might be useful to shorten the code a bit.

Reply:
Agreed offline as a follow-up PR for the 1.6 release.